hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/BaseTokenHandler.java

```diff
@@ -423,22 +428,34 @@ public static boolean isExpiryOnlyUpdateOp(@NonNull final TokenUpdateTransaction
     }
 
     /**
-     * Returns true if the given token update op is a metadata-only update op.
-     * This is needed for validating whether a token update op has admin key present on the token,
-     * to update any other fields other than metadata.
-     * For updating metadata we need signature from either admin key or metadata key
-     * @param op the token update op to check
-     * @return true if the given token update op is an metadata-only update op
+     * Check if TokenUpdateOp wants to update only some of the low priority keys or the metadata field
+     * low priority keys are -> wipeKey, kycKey, supplyKey, freezeKey, feeScheduleKey, pauseKey or metadataKey
      */
-    public static boolean isMetadataOnlyUpdateOp(@NonNull final TokenUpdateTransactionBody op) {
-        if (!op.hasMetadata()) {
-            return false;
+    public static boolean noOtherFieldThanLowPriorityKeyOrMetadataWillBeUpdated(
+            @NonNull final TokenUpdateTransactionBody op) {
+        return !((op.symbol() != null && !op.symbol().isEmpty())
+                || (op.name() != null && !op.name().isEmpty())
+                || op.hasTreasury()
+                || op.hasAdminKey()
+                || op.hasAutoRenewAccount()
+                || op.hasAutoRenewPeriod()
+                || op.hasMemo()
+                || op.hasExpiry());
+    }
+
+    /**
+     * Check if a given token already has some of the low priority keys
+     */
+    public static boolean hasAlreadySomeNonAdminKeys(@NonNull final Token token) {
+        // here we want to remove the admin key case because we need only low priority keys
+        final var nonAdminKeys =
```

Comment:
Instead of re-creating this every time, I think just define above:
```
private static final Set<TokenKeys> NON_ADMIN_TOKEN_KEYS = EnumSet.complementOf(EnumSet.of(TokenKeys.ADMIN_KEY));
```